dags/als-kp-dataset-dag.py

```
@@ -221,103 +332,157 @@ def transform_data(data: Dict[str, Any], **context) -> List[Dict[str, Any]]:
+        """Get all existing ALS datasets from the collection, grouped by ALS number.
         Returns:
-            str: A formatted Slack message string describing the duplicate datasets.
+            Dict mapping als_number -> dataset_info with fields:
+            - synapse_id, dataset_code, prefix, priority, date_str, sameAs
         """
-        if not duplicates:
-            return ""
-        message = "There are datasets that need to be reviewed: \n\n"
-        for index, item in enumerate(duplicates):
-            printed_message = f"{index+1}. C-path identifier: {item['sameAs']}, title: {item['title']}, creator: {item['creator']}, keywords: {item['keywords']}, subject: {item['subject']}, collection: {item['collection']}, publisher: {item['publisher']}, species: {item['species']}, url: {item['url']}, description: {item['description']} \n\n"
-            message += printed_message
-        return message
+        synapse_client = syn_hook.client
+        query_str = f"SELECT * FROM {collection_id} WHERE publisher='Critical Path Institute'"
+        current_data = synapse_client.tableQuery(query_str).asDataFrame()
+
+        existing_datasets = {}
+
+        for _, row in current_data.iterrows():
+            url = row.get('url', '')
+            dataset_code = extract_dataset_code_from_url(url)
+
+            if not dataset_code:
+                continue
+
+            try:
+                als_number, prefix, date_str = parse_dataset_code(dataset_code)
+                existing_datasets[als_number] = {
+                    'synapse_id': row['id'],
+                    'dataset_code': dataset_code,
+                    'prefix': prefix,
+                    'priority': get_version_priority(prefix),
+                    'date_str': date_str,
+                    'sameAs': row['sameAs'],
+                    'url': row['url']
+                }
+            except ValueError:
+                continue
```

Be careful swallowing exceptions without logging what the underlying failure was. In this case this could silently fail